projector-html-haskell/src/Projector/Html/Backend/Haskell/Syntax.hs

```diff
@@ -0,0 +1,264 @@
+{-| Some shorthand for constructing Haskell98-ish TH syntax trees.
```

Comment:
Not sure how much of this is used, but it was enough for me to just pull this all in for now - not sure how useful it is for a single project - can probably just be inlined at some point.